packages/node-sdk/README.md

```diff
@@ -0,0 +1,170 @@
+# Bucket Node.js SDK
+
+Node.js, JavaScriptS/Typescript feature flag and tracking client for [Bucket.co](https://bucket.co).
+
+## Installation
+
+Install using `yarn` or `npm` with:
+
+> `yarn add -s @bucketco/node-sdk` or `npm install -s @bucketco/node-sdk`.
+
+Other languages than JavaScript/Typescript are currently not supported by an SDK.
+You can [use the HTTP API directly](https://docs.bucket.co/reference/http-tracking-api)
+
+## Basic usage
+
+Before the library can be used, you need to obtain a secret key from
+[Environment setting view](https://app.bucket.co/envs/{environment}/settings/app-environments)
+in **Bucket.co**.
+
+> Note that secret keys should only be used by the `node-sdk` and not any other
+> SDKs provided by Bucket.Secret keys offer the users the ability to obtain
+> information that is often sensitive and thus should not be used in any
+> front-end application.
+
+```ts
+import { Client } from "@bucketco/node-sdk";
+
+// Create a new instance of the client with the secret key. Additional options
+// are available, such as supplying a logging sink, a custom HTTP client and
+// other custom properties.
+//
+// We recommend that only one global instance of `client` should be created
+// to avoid multiple round-trips to our servers.
+const client = new Client({ secretKey: "sec_prod_xxxxxxxxxxxxxxxxxxxxx" });
+
+// Initialize the client's feature flag retrieval loop. This is not a mandatory
+// call, but it is recommended in order to warm up the cache so that future
+// calls to `getFlags()` will have cached flag definitions available.
+//
+// You can also supply your `fallbackFlags` to the `initialize()` method. These
+// fallback flags are used in the situation when the server-side flags are not
+// yet loaded or there are issues retrieving them.
+await client.initialize();
+```
+
+At any point where the client needs to be used, we can configure it through
+`withUsr()`, `withCompany()` and `withCompanyContext()` methods. Each of
+these three methods returns a new instance of `Client` class which includes
+the applied user, company or context:
+
+```ts
+// configure the client
+const boundClient = client
+  .withUser({ id: "john_doe" name: "John Doe" })
+  .withCompany({ id: "acme_inc", name "Acme Inc."})
+  .withCustomContext({ additional: "context", number: [1,2,3] })
```

Comment:
Do these chains result in a new immutable instance for each call, or does the first one create a new instance which is then mutated by the remaining calls?

Only could easily expect user, company and context to be available in server middlewares at different points, so it won't be possible to set up this chain once and for all in a single place. Rather these pieces of information will have to be added piecemeal. If a new immutable instance is created each time, the client reference will have to be re-assigned in each place. It might be relevant to show an example of this.

If I wanted to check a flag without any user, company or custom context, would I just access the global client instance? I'm thinking of when we presumably add always on/off rules per environment. These could be evaluated fine without any additional context other than the secret key

Reply:
We discussed and came up with changes.